### DISCUSSION TITLE
BeEquivalentTo assertion is failing without telling me the exact mismatch on the property it failed on for large object

### DISCUSSION THREAD

**Discussion by Morshed11218** [fluentassertions/fluentassertions#2187] [General]
I'm noticing that when I'm comparing two large object with different class type using `BeEquivalentTo` assertion, the failure message just says that the actual object does not match expected object. It's not telling me exactly on what property the mismatch is occurring on. 

I see this failure when I upgrade FluentAssertion package from 4.19.4 to 6.10.0

Assertion code:
```cs
actualIndication.Should().BeEquivalentTo(expected, opt => opt
    .ExcludingMissingMembers()
    .Excluding(info => info.InvestorAccountId)
    .Excluding(l => l.CreatedDateTime)
    .Excluding(i => i.LastModifiedDateTime));
```

This is the error log I'm getting:
[Indication_search_by_dealId_should_return_expected_results-638174189832304262.txt](https://github.com/fluentassertions/fluentassertions/files/11264368/Indication_search_by_dealId_should_return_expected_results-638174189832304262.txt)

As you can see from the error log, it doesn't exactly tell me where the issue is.

Here is comparison image of the two object. Left is expected and right is actual. As you can see from the image, I have taken care of all the fields that has a mismatch (InvestorAccountId, CreatedDateTime, and LastModifiedDateTime nad ExcludingMissingMembers) but it still fails. 

![Gmail-Your-comparison-from-Text-Compare_](https://user-images.githubusercontent.com/2136499/232854953-33518359-6bf0-49d6-a2cc-f4ec64df51ea.jpg)

**Comment by dennisdoomen** [comment]
v5 and v6 had a lot of breaking changes, especially around `BeEquivalentTo` and how the expectation now drives the assertion. Check out the [upgrade notes for v5](https://www.continuousimprover.com/2018/02/fluent-assertions-50-best-unit-test.html#upgrading-tips) and [those for v6](https://fluentassertions.com/upgradingtov6). It could be that you need to swap the two objects you're comparing. It could also that your types override `Equals` or records are involved.

**Comment by Morshed11218** [comment]
Found the issue. My expected class was overriding the `IEquatable.Equals()`. Fixed the issue by having options: `ComparingByMembers`

ex:
```cs
actualIndications.Should().BeEquivalentTo(expectedIndications, opt => opt
    .ExcludingMissingMembers()
    .ComparingByMembers<Ipreo.NS.Indication.Domain.Model.Indication.Indication>()
    .Excluding(info => info.InvestorAccountId)
    .Excluding(l => l.CreatedDateTime)
    .Excluding(i => i.LastModifiedDateTime));
```
